src/Main.java

```diff
@@ -107,9 +107,15 @@ public static void main(String[] args) {
                     System.out.println("Enter String (Maximal Square challenge)");
                     str = scanner.nextLine();
 
-                    int matrix[][] = matrix(str);
+                    int[][] matrix = matrix(str);
```

Comment:
You asked here is the difference

They are semantically identical. The int array[] syntax was only added to help C programmers get used to java.
int[] array is much preferable and less confusing.

Reply:
thank you for info